usr/lib/systemd/system/greenboot-grub2-set-counter.service

```diff
@@ -0,0 +1,21 @@
+#  SPDX-License-Identifier: LGPL-2.1+
+#
+#  This file is part of greenboot.
+#
+#  greenboot is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Set grub2 boot counter in preparation of upgrade
+Before=ostree-finalize-staged.service
+ConditionPathExists=/run/ostree/staged-deployment
```

Comment:
@jlebon is this file here when we install or override arbitrary packages? I think this way we can make sure we only set the grub countdown when we actually need to.

Reply:
Yeah, it'll exist anytime there's a new deployment we created. To contrast, note that e.g. `rpm-ostree rollback` doesn't create a deployment but just re-arranges existing ones.

Reply:
(Just had a look at the service file and it looks good overall!)